lib/improver/tests/test_percentile_percentileconverter.py

```diff
@@ -0,0 +1,208 @@
+
+        """
+        plugin = PercentileConverter(collapse_coord)
+        result = plugin.process(self.cube)
+
+        # Check percentile values.
+        self.assertArrayAlmostEqual(
+            result.data[:, 0, 0], [0., 0., 1., 2., 2., 3., 4., 5., 6., 7., 8.,
+                                   8., 9., 10., 10.])
+        # Check coordinate name.
+        self.assertEqual(result.coords()[0].name(),
+                         'percentile_over_latitude_longitude')
+        # Check coordinate points.
+        self.assertArrayEqual(
+            result.coord('percentile_over_latitude_longitude').points,
+            [0, 5, 10, 20, 25, 30, 40, 50, 60, 70, 75, 80, 90, 95, 100])
+        # Check resulting data shape.
+        self.assertEqual(result.data.shape, (15, 3, 1))
+        # Check demoted dimension coordinates exists as scalars with bounds.
+        self.assertArrayEqual(result.coord('longitude').bounds,
+                              [[-180., 180.]])
+        self.assertArrayEqual(result.coord('latitude').bounds, [[-90., 90.]])
+
+    def test_valid_single_coord_string(self):
+        """
+        Test that the plugin handles a valid collapse_coord passed in
+        as a string.
+
+        """
+        collapse_coord = 'longitude'
+        self.passed_single_coordinate(collapse_coord)
+
+    def test_valid_single_collapse_coord_dimcoord(self):
+        """
+        Test that the plugin handles a valid collapse_coord passed in
+        as a DimCoord.
+
+        """
+        collapse_coord = self.longitude
+        self.passed_single_coordinate(collapse_coord)
+
+    def test_valid_multi_coord_string_list(self):
+        """
+        Test that the plugin handles a valid list of collapse_coords passed in
+        as a list of strings.
+
+        """
+        collapse_coord = ['longitude', 'latitude']
+        self.passed_multiple_coordinates(collapse_coord)
+
+    def test_valid_multi_coord_dimcoord_list(self):
+        """
+        Test that the plugin handles a valid list of collapse_coords passed in
+        as a list of DimCoords.
+
+        """
+        collapse_coord = [self.longitude, self.latitude]
+        self.passed_multiple_coordinates(collapse_coord)
+
+    def test_valid_multi_coord_mixed_list(self):
+        """
+        Test that the plugin handles a valid list of collapse_coords passed in
+        as a list of DimCoords and strings.
+
+        """
+        collapse_coord = ['longitude', self.latitude]
+        self.passed_multiple_coordinates(collapse_coord)
+
+    def test_unavailable_collapse_coord(self):
+        """
+        Test that the plugin handles a collapse_coord that is not
+        available in the cube.
+
+        """
+        collapse_coord = 'not_a_coordinate'
+        plugin = PercentileConverter(collapse_coord)
+        msg = "Coordinate "
+        with self.assertRaisesRegexp(CoordinateNotFoundError, msg):
+            plugin.process(self.cube)
+
+    def test_invalid_collapse_coord_type(self):
```

Comment:
This test should be on the `_test_type` function, rather than being tested as part of the call to `process.`